best_practices.md

```diff
@@ -0,0 +1,271 @@
+# Best practices for programming diagnostics
+
+To facilitate code integration, code sharing between developers, reusability, easy
+maintenance, and readability, it is recommended that these recommendations
+are followed when developing new diagnostic scripts.
+
+It is recommended that you contribute your diagnostic script to an existing
+tool, such as the [ESMValTool](https://www.esmvaltool.org), as this makes it easier
+to find users and collaborators, thus increasing the chances that your software will
+survive long after you have moved on to something new and exciting.
+However, if you find that contributing to existing software does not work for you,
+you can start a new package.
+A convenient way to do so is by using a [cookiecutter](https://cookiecutter.readthedocs.io)
+template that already implements many of the items recommended here, for example
+this [python-template](https://github.com/NLeSC/python-template).
+Regardless of whether you are contributing to an existing project or starting
+a new project, it is recommended that you follow the guidelines in this document.
+
+## Make your code open-source
+
+It is recommended that you make your code
+[open-source](https://the-turing-way.netlify.app/reproducible-research/open/open-source.html)
+from the start, to facilitate reproducible science and encourage potential
+[collaboration](https://the-turing-way.netlify.app/collaboration/collaboration.html).
+If you have reservations about making your code open source before publication,
+consider developing it in a private [git repository](#use-git-for-version-control) and making it
+public once you have submitted your paper.
+
+## Add a license
+
+It is critical that you share your code with a
```

Comment:
Can a recommendation be qualified as 'critical' ?

Reply:
Transferred to #8.